tools/kprint/main.cpp

```diff
@@ -1,33 +1,36 @@
 #include "kllvm/ast/AST.h"
 #include "kllvm/printer/printer.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/WithColor.h"
```

Comment:
What does this header pull in? I recommend running [IWYU](https://include-what-you-use.org/) over your code to make sure your includes are correct.

Reply:
I found out that it asks to include more than the necessary libraries. Should I include them?

```
../tools/kprint/main.cpp should add these lines:
#include <__functional/function.h>       // for __base
#include "llvm/ADT/StringMap.h"          // for StringMap
#include "llvm/ADT/StringRef.h"          // for StringRef
#include "llvm/Support/ErrorHandling.h"  // for llvm

../tools/kprint/main.cpp should remove these lines:
- #include <string>  // lines 7-7
- #include "llvm/Support/WithColor.h"  // lines 4-4

The full include-list for ../tools/kprint/main.cpp:
#include <__functional/function.h>       // for __base
#include <unistd.h>                      // for isatty
#include <iostream>                      // for string
#include "llvm/ADT/StringMap.h"          // for StringMap
#include "llvm/ADT/StringRef.h"          // for StringRef
#include "llvm/Support/CommandLine.h"    // for opt, desc, init, Positional
#include "llvm/Support/ErrorHandling.h"  // for llvm
```

Reply:
It can be a bit finicky to get it to run properly, so take the suggestions with a pinch of salt - e.g. `<__functional/function.h>` probably isn't one we need. It's a sanity check rather than a prescriptive guide